crates/polars-time/src/replace.rs

```diff
@@ -0,0 +1,187 @@
+use polars_core::prelude::*;
+
+use crate::prelude::*;
+
+#[cfg(feature = "dtype-datetime")]
+pub trait PolarsReplaceDatetime: DatetimeMethods {
+    #[allow(clippy::too_many_arguments)]
+    fn replace(
+        &self,
+        year: &Int32Chunked,
+        month: &Int8Chunked,
+        day: &Int8Chunked,
+        hour: &Int8Chunked,
+        minute: &Int8Chunked,
+        second: &Int8Chunked,
+        microsecond: &Int32Chunked,
+        ambiguous: &StringChunked,
+    ) -> PolarsResult<Self>
+    where
+        Self: Sized;
+}
+
+#[cfg(feature = "dtype-date")]
+pub trait PolarsReplaceDate: DateMethods {
+    fn replace(
+        &self,
+        year: &Int32Chunked,
+        month: &Int8Chunked,
+        day: &Int8Chunked,
+    ) -> PolarsResult<Self>
+    where
+        Self: Sized;
+}
+
+#[cfg(feature = "dtype-datetime")]
+impl PolarsReplaceDatetime for DatetimeChunked {
+    #[allow(clippy::too_many_arguments)]
+    fn replace(
+        &self,
+        year: &Int32Chunked,
+        month: &Int8Chunked,
+        day: &Int8Chunked,
+        hour: &Int8Chunked,
+        minute: &Int8Chunked,
+        second: &Int8Chunked,
+        microsecond: &Int32Chunked,
+        ambiguous: &StringChunked,
+    ) -> PolarsResult<Self> {
+        let n = self.len();
+
+        // For each argument, we must check if:
+        // 1. No value was supplied (None)       --> Use existing year from Series
+        // 2. Value was supplied and is a Scalar --> Create full Series of value
+        // 3. Value was supplied and is Series   --> Update all elements with the non-null values
+        let year = if year.len() == 1 {
+            // SAFETY: array has one value.
+            if let Some(value) = unsafe { year.get_unchecked(0) } {
```

Comment:
No need to use unsafe for a single value.

Reply:
Fixed in [34d39ea](https://github.com/pola-rs/polars/pull/19708/commits/34d39ea3b0613036bd988dbb5ce9d6cbab2dcd67). I'll resolve all of these.